packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/interfaces/query_platform.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:async';
+
+import 'package:meta/meta.dart' show required;
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import '../types.dart';
+
+/// The Query platform interface.
+abstract class QueryPlatform extends PlatformInterface {
+  /// Constructor
+  QueryPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  /// The default instance of [QueryPlatform] to use.
+  ///
+  /// Platform-specific plugins should override this with their own class
+  /// that extends [QueryPlatform] when they register themselves.
+  static QueryPlatform get instance => _instance;
+
+  static QueryPlatform _instance;
+
+  // TODO(amirh): Extract common platform interface logic.
+  // https://github.com/flutter/flutter/issues/43368
+  static set instance(QueryPlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  // Actual interface
```

Comment:
I am leaning towards removing this non-Dartdoc comment (in all `PlatformInterface` subclasses) since people will be follow the example of this plugin and it clutters up the class a bit. If the code is self-documenting only Dartdoc comments will be necessary.